### DISCUSSION TITLE
When calling the app.run() method in the main function, it blocks the thread as expected. However, when running it in a thread, it doesn't block, which causes the HTTP service to fail.

### DISCUSSION THREAD

**Discussion by LetAmericaGreatAgain** [uNetworking/uWebSockets#1817] [Q&A]
When calling the app.run() method in the main function, it blocks the thread as expected. However, when running it in a thread, it doesn't block, which causes the HTTP service to fail.

```c++
std::thread daemon([this]() {
        this->app->run();
    });
    daemon.join();
```

**Comment by Jacob-Burckhardt** [comment]
Your code mentions the `app` variable.  You create `app` on one thread and you pass it to another thread via the lambda capture.  Instead, try creating `app` in the `daemon` thread.

**Comment by uNetworkingAB** [comment]
It doesn't block because there is nothing to run. When you call run in another thread, it will use that thread's loop. But you have registered stuff in the main thread's loop.

run() uses thread local Loop. So you are trying to run another loop. You need to register stuff on the same thread that runs the app. Or, simply, you just need to stop using threads, since the library is single-threaded (with exceptions)
